lib/facilities/local_store.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+require 'facilities/bulk_client'
+
+module Facilities
+  class LocalStore
+    attr_reader :adapter
+
+    GIS_CHECK_FREQUENCY = 1.hour
+    BATCH_SIZE = 1000.0
+    OPEN_TIMEOUT = 2
+    REQUEST_TIMEOUT = 6
+    Pointer = Struct.new(:x, :y, :facility)
+
+    def initialize(url, _type, adapter)
+      @url = url
+      @adapter = adapter
+      @mutex = Mutex.new
+      @last_check = Time.new(0).utc
+      @current_gis_update = 0
+      @client = BulkClient.new(@url)
+      @index = {}
+      @coordinates = []
+      # check_for_freshness
+    end
+
+    def get(id)
+      check_for_freshness
+      @mutex.synchronize do
+        return @index[id]
+      end
+    end
+
+    def query(bbox, services = nil)
+      check_for_freshness
+      filter = services ? @adapter.with_services(services) : ->(_) { true }
```

Comment:
I'm finding this with_services thing to be extremely difficult to parse. It seems like another case where the non-idiomatic Ruby is revealing challenges in the encapsulation of the code. I'm a bit too confused to figure out what the encapsulation should be in this case though.